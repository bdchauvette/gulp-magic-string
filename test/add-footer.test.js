const { addFooter } = require("..")
const { fibonacci } = require("./fixtures")
const { transform } = require("./helpers")

describe("string footer", () => {
  it("adds the provided string to the end of the file contents", async () => {
    const file = fibonacci.toVinyl()
    const transformedFile = await transform(file, addFooter("ğŸ‘£ğŸ‘£ğŸ‘£"))
    expect(transformedFile.contents.toString()).toMatchSnapshot()
  })
})

describe("function footer", () => {
  it("passes the file to the function", async () => {
    const file = fibonacci.toVinyl()
    const footer = jest.fn()
    await transform(file, addFooter(footer))
    expect(footer).toHaveBeenCalledTimes(1)
    expect(footer).toHaveBeenLastCalledWith(file)
  })

  it("adds the generated footer to the end of the file contents", async () => {
    const file = fibonacci.toVinyl()
    const footer = (f) => `ğŸ‘£ğŸ‘£ğŸ‘£ - ${f.basename}`
    const transformedFile = await transform(file, addFooter(footer))
    expect(transformedFile.contents.toString()).toMatchSnapshot()
  })
})
