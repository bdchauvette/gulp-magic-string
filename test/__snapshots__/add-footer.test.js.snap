// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function footer adds the generated footer to the end of the file contents 1`] = `
"export function fibonacci(num) {
  if (num > 1) {
    return fibonacci(num - 1) + fibonacci(num - 2)
  }

  return 1
}

ğŸ‘£ğŸ‘£ğŸ‘£ - fibonacci.js"
`;

exports[`string footer adds the provided string to the end of the file contents 1`] = `
"export function fibonacci(num) {
  if (num > 1) {
    return fibonacci(num - 1) + fibonacci(num - 2)
  }

  return 1
}

ğŸ‘£ğŸ‘£ğŸ‘£"
`;
